pkg/client/unversioned/request.go

```diff
@@ -839,7 +839,10 @@ func isTextResponse(resp *http.Response) bool {
 // checkWait returns true along with a number of seconds if the server instructed us to wait
 // before retrying.
 func checkWait(resp *http.Response) (int, bool) {
-	if resp.StatusCode != errors.StatusTooManyRequests {
+	switch r := resp.StatusCode; {
+	// any 500 error code and 429 can trigger a wait
+	case r == errors.StatusTooManyRequests, r >= 500:
```

Comment:
I liked what I saw until I got here. Why should our client retry on server errors?  Can you tighten this to "500's that are also ServerTimeout errors"?  That would require pulling the body.